homeassistant/components/acmeda/helpers.py

```diff
@@ -0,0 +1,45 @@
+"""Helper functions for Acmeda Pulse."""
+from homeassistant import config_entries
+from homeassistant.helpers.device_registry import async_get_registry as get_dev_reg
+from homeassistant.helpers.entity_registry import async_get_registry as get_ent_reg
+
+from .const import DOMAIN
+
+
+async def remove_devices(hass, config_entry, api_ids, current):
+    """Get items that are removed from api."""
+    removed_items = []
+
+    for item_id in current:
+        if item_id in api_ids:
+            continue
+
+        # Device is removed from Pulse, so we remove it from Home Assistant
+        entity = current[item_id]
+        removed_items.append(item_id)
+        await entity.async_remove()
+        ent_registry = await get_ent_reg(hass)
```

Comment:
await entity.async_remove() in `helpers/entity.py` is doing this under the hood.

It looks like much of the below can be removed.

```

    async def async_remove(self) -> None:
        """Remove entity from Home Assistant."""
        assert self.hass is not None
        await self.async_internal_will_remove_from_hass()
        await self.async_will_remove_from_hass()

        if self._on_remove is not None:
            while self._on_remove:
                self._on_remove.pop()()

        self.hass.states.async_remove(self.entity_id, context=self._context)


```

Reply:
I couldn't work out how to have the entity completely removed from the user interface without this code. The code was copied from the hue integation in `components/hue/helpers/py`.